imageio/plugins/tifffile_v3.py

```diff
@@ -302,9 +302,12 @@ def metadata(
                 if flag_value and hasattr(self._fh, flag + "_metadata"):
                     flavor_metadata = getattr(self._fh, flag + "_metadata")
                     if isinstance(flavor_metadata, tuple):
-                        metadata.update(flavor_metadata[0])
-                    else:
+                        flavor_metadata = flavor_metadata[0]
+                    if isinstance(flavor_metadata, dict):
                         metadata.update(flavor_metadata)
+                    else:
+                        # ome only?
+                        metadata.update({flag: flavor_metadata})
```

Comment:
Could you explain your logic here so I can follow along? 

If the `else` is only relevant for OME then I think we should switch it around and do something like:

```python
if flag == 'ome':
    ...  # ome-specific metadata loading
elif isinstance(flavor_metadata, tuple):
    ...  # tuple-specific metadata loading
else:
    ...  # normal metadata loading
```

Reply:
I even considered ignoring non-dict entries. But it makes it slightly non-trivial to extract XML because file-level "description" is missing somehow, so one would have to write `iio.immeta(filename, index=0)` to get it.